captum/attr/_core/neuron/neuron_integrated_gradients.py

```diff
@@ -30,17 +40,45 @@ def __init__(self, forward_func, layer, device_ids=None):
         NeuronAttribution.__init__(self, forward_func, layer, device_ids)
         GradientAttribution.__init__(self, forward_func)
 
+    @typing.overload
     def attribute(
         self,
-        inputs,
-        neuron_index,
-        baselines=None,
-        additional_forward_args=None,
-        n_steps=50,
-        method="gausslegendre",
-        internal_batch_size=None,
-        attribute_to_neuron_input=False,
-    ):
+        inputs: Tensor,
+        neuron_index: Union[int, Tuple[int, ...]],
+        baselines: Optional[Union[Tensor, Tuple[Tensor, ...]]] = None,
+        additional_forward_args: Any = None,
+        n_steps: int = 50,
+        method: str = "gausslegendre",
+        internal_batch_size: Optional[int] = None,
+        attribute_to_neuron_input: bool = False,
+    ) -> Tensor:
+        ...
+
+    @typing.overload
+    def attribute(
+        self,
```

Comment:
It looks like we have multiple copies of `def attribute`  and duplicated code. I think that the original solution was probably better. Why did we decided to change it ?

Reply:
The original constructor / type hints are still there, this just adds variants that show that the return type will be a Tensor when a Tensor input is given, etc. The reason this was needed for Integrated Gradients was that the return type is quite complex, and this causes errors when type checking code calling IG that assumes a particular return type, while the generic one contains a union of a variety of options. The callers generally assume one of the known behaviors, e.g. providing a single tensor with no compute_convergence_delta flag always returns a single tensor. By including this, the calling code can be type checked with the appropriate return type as opposed to the union of all possible return types. This was causing issues in both insights as well as neuron IG, which is why I added it to IG as well as here. Unfortunately, it does require duplicating the method signature (but at least none of the implementation, so it's just a decoration that doesn't affect runtime), so in general, it might be good to include these signatures whenever possible, since they help to type check code using Captum methods more accurately.

Reply:
For now, I've left the original type hints as well as the overload signatures to be consistent with other methods, but we can consider removing the main definition type hints with unions and just leaving the overload signatures if that looks cleaner?